ssllib.c

```diff
@@ -307,6 +307,23 @@ ssl_strdup(const char *s)
     return new;
 }
 
+static RSA *
+ssl_rsadup(const RSA *rsa)
+{
+    RSA *c = RSA_new();
+
+    c->n = rsa->n;
```

Comment:
I have augmented the pull request, please have a look: I have added the error checking, and am now performing a deep copy of the struct members, which is I think the proper way to copy these fields (and only those that we actually set). Thank you!